File: 94-binary-tree-inorder-traversal/binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> group=new ArrayList<>();
        inorder(root,group);
        return group;

    }
    public void inorder(TreeNode node,List<Integer> group)
    {
        if(node==null) return;
        inorder(node.left,group);
        group.add(node.val);
        inorder(node.right,group);

    }
}

File: 94-binary-tree-inorder-traversal/binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        st=[]
        res=[]
        cur=root
        while cur!=None or st!=[]:
            while cur!=None:
                st.append(cur)
                cur=cur.left
            node=st.pop()
            res.append(node.val)
            cur=node.right
        return res
        